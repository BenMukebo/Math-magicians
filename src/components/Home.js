import React from 'react';

const Home = () => (
  <section className="home-section">
    <h2>
      Welcome To The Best
      <span> Calculator </span>
      Out There!
    </h2>
    <p className="">
      Calculators have not only greatly enhanced our ability to perform the regular
      computations that are involved in everyday life, but provided humans with the ability
      to understand mathematics on a greater scale than ever imagined.
      Calculations which were previously cumbersome and time-consuming can now be done in minutes or
      even seconds, all at the push of a few buttons. Without calculators, advanced math courses,
      such as Calculus, would require much longer class-times and reformatted lesson plans.
      Calculators, when used alongside a comprehensive mathematics curriculum,
      increase the quality and quantity of student learning
    </p>
    <p className="">
      This technology allows students solve complicated problems quickly and in an efficient manner.
      Additionally, it can reduce the problem to simpler tasks and allows the student to devote
      more time in understanding the problem. Secondly, they are saved from monotonous calculations
      and the same boring mundane procedure. One type of number trick involves adding, subtracting,
      multiplying, or dividing by a starting number and subtracting the original number in such
      a way that each participant always arrives at a certain number: Think of a number between
      1 and 100. Multiply your number by 4. Add 12.
    </p>
  </section>
);

export default Home;
