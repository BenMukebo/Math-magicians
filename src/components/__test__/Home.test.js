import React from 'react';
import renderer from 'react-test-renderer';
// import isTSAnyKeyword from '@babel/types';
// import { render } from '@testing-library/react';
import Home from '../Home';

it('matches snapshot', () => {
  const tree = renderer.create(
    <section className="home-section">
      <h2 className="second-title">
        Welcome To The Best
        <span> Calculator </span>
        In The world!
      </h2>
      <div className="paragraph">
        <p>
          Calculators have not only greatly enhanced our ability to perform the regular
          computations that are involved in everyday life, but provided humans with the ability
          to understand mathematics on a greater scale than ever imagined.
        </p>
        <p>
          This technology allows students solve complicated problems quickly and in an efficient
          manner.
        </p>
      </div>
      <button className="btn-more" type="button">Read More</button>
    </section>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
