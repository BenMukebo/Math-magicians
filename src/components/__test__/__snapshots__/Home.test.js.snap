// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<section
  className="home-section"
>
  <h2
    className="second-title"
  >
    Welcome To The Best
    <span>
       Calculator 
    </span>
    In The world!
  </h2>
  <div
    className="paragraph"
  >
    <p>
      Calculators have not only greatly enhanced our ability to perform the regular computations that are involved in everyday life, but provided humans with the ability to understand mathematics on a greater scale than ever imagined.
    </p>
    <p>
      This technology allows students solve complicated problems quickly and in an efficient manner.
      <span
        className="show"
      >
        Students can develop fluency and mental math skills without a calculator. Visual models and number sentences help students find answers while also developing number sense, mental math, and problem solving abilities. When students become reliant on calculators, their fluency and number sense actually decline over time.
      </span>
    </p>
  </div>
  <button
    className="btn-more"
    type="button"
  >
    Read More
  </button>
</section>
`;
